Source/JavaScriptCore/parser/NodesAnalyzeModule.cpp

```diff
@@ -32,18 +32,24 @@
 
 namespace JSC {
 
-static Expected<RefPtr<ScriptFetchParameters>, String> tryCreateAssertion(VM& vm, ImportAssertionListNode* assertionList)
+static Expected<RefPtr<ScriptFetchParameters>, std::tuple<ErrorType, String>> tryCreateAttributes(VM& vm, ImportAttributesListNode* attributesList)
 {
-    if (!assertionList)
+    if (!attributesList)
         return RefPtr<ScriptFetchParameters> { };
 
+    // https://tc39.es/proposal-import-attributes/#sec-AllImportAttributesSupported
     // Currently, only "type" is supported.
     std::optional<ScriptFetchParameters::Type> type;
-    for (auto& [key, value] : assertionList->assertions()) {
+    for (auto& [key, value] : attributesList->attributes()) {
+        if (*key != vm.propertyNames->type)
+            return makeUnexpected(std::tuple { ErrorType::SyntaxError, makeString("Import attribute \""_s, StringView(key->impl()), "\" is not supported"_s) });
+    }
+
+    for (auto& [key, value] : attributesList->attributes()) {
         if (*key == vm.propertyNames->type) {
```

Comment:
Why is this check needed?  Isn't this already proven to be always true in line 44 above?

Reply:
Right now, we intentionally do this to make it available to support more attributes easily.